src/dataregistry/registrar_util.py

```diff
@@ -68,17 +69,24 @@ def _form_dataset_path(owner_type, owner, relative_path, root_dir=None):
         Owner of dataset
     relative_path : str
         Relative path within the data registry
-    root_dir : str
+    schema : str, optional
+        Schema we are connected to
+    root_dir : str, optional
         Root directory of data registry
+    dialect : str, optional
+        SQL dialect, e.g postgres or sqlite
 
     Returns
     -------
     to_return : str
-        Full path of dataset in the data registry
+        Full (or relative) path of dataset in the data registry
     """
+
     if owner_type == "production":
         owner = "production"
     to_return = os.path.join(owner_type, owner, relative_path)
+    if schema:
```

Comment:
It doesn't make sense to call this routine with schema argument == None and root_dir != None. It's an internal routine which currently is always called properly, but wouldn't hurt to raise an exception in the bad case, or at least put something in the docstring.